frontend/src/map/map-generation.ts

```diff
@@ -0,0 +1,125 @@
+export class MapGenerator {
+  
+  #map: number[][] = [];
+  
+  #n: number;
+
+  #m: number;
+
+  constructor(n: number, m: number) {
+    this.#n = n;
+    this.#m = m;
+    this.#map = [];
+  }
+
+  create_empty_map() {
```

Comment:
#createEmptyMap() : number[][] {

lo haria privado

Reply:
podes hacer algo cheto como Array.from({lengtth: this.#row}, () => Array(this.#column).fill(0))

es bastante mas lindo que un for. por lo general en js moderno es raro hacer un for de la manera tipica

[](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/from#arraylike)